### PR TITLE
[DPE-4976] Run tests on juju 3.6 on a nightly schedule

### PR DESCRIPTION
**Pull request by shayancanonical**
## Issue
We would like to run nightly tests with juju 3.6

## Solution
Upgrade to dpw v21.0.0 and add juju 3.6 into the matrix for tests. Also, upgrade libjuju to ^3.5.2.0

### CODE REVIEW

**Comment by shayancanonical** [comment]
Tests pass with juju 2.9/stable, 3.4/stable and 3.6/beta in https://github.com/canonical/mysql-router-operator/actions/runs/10527081121

**Review by taurus-forever — APPROVED** [review]
LGTM if series works on 3.6

**Review by paulomach — APPROVED** [review]

File: tests/integration/test_subordinate_charms.py [hunk 1]
```diff
@@ -46,6 +47,7 @@ async def test_ubuntu_pro(ops_test, mysql_router_charm_series, github_secrets):
             application_name=UBUNTU_PRO_APP_NAME,
             channel="latest/edge",
             config={"token": github_secrets["UBUNTU_PRO_TOKEN"]},
+            series=mysql_router_charm_series,
```
[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/test_subordinate_charms.py:50]: even though series is deprecated, i believe support for series still exists in `juju=3.5.2.0` [reference](https://github.com/juju/python-libjuju/blob/3.5.2.0/juju/model.py#L1672-L1690)

[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/test_subordinate_charms.py:50]: I believe we could use either. In this repo, fixtures and CI is deeply tied to series usage. Switching to bases would be a bigger undertaking than this PR

[the remark this answers is not in the document]
Comment (shayancanonical) [reply tests/integration/test_subordinate_charms.py:50]: ack, just confirmed that using `--series` works as expected when deploying in a juju 3.6-beta2 model

**Pull request merged**